qualtran/_infra/binst_graph_iterators.py

```diff
@@ -0,0 +1,100 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import heapq
+from collections import Counter
+from typing import Dict, Iterator, List, TYPE_CHECKING
+
+import attrs
+import networkx as nx
+
+if TYPE_CHECKING:
+    from qualtran import BloqInstance
+
+_INFINITY: int = int(1e18)
```

Comment:
Eventually, I want to set `_INFINITY + binst.i` as the priority instead of `_INFINITY`; assuming we can make `binst.i` preserve insertion order following https://github.com/quantumlib/Qualtran/issues/1098

Reply:
In that case, a name other than "infinity" should be used if we're going to literally do "infinity + 1" as a meaningful distinction :p

Reply:
renamed